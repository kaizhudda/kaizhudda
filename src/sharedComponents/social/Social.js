import React from 'react';

import './Social.scss';

const Social = ({ className }) => {
  return (
    <p className={`Social ${className ? className : ''}`}>
      <a href=""><i className="fab fa-linkedin" /></a>
      <a href=""><i className="fab fa-github-square" /></a>
      <a href=""><i className="fab fa-twitter-square" /></a>
    </p>
  );
}

export default Social;
