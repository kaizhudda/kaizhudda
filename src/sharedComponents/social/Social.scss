@import '../../styles/variables';
@import '../../styles/media-queries';

.Social {
  p {
    margin: 10px 0;
  }
  .fab {
    // color: #22a39f;
    font-size: 2rem;
    margin-right: 15px;
    color: white;

    &:hover {
      color: #efefef;
    }
  }
}