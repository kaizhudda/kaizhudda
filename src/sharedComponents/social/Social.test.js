import React from 'react';
import { mount } from 'enzyme';
import Social from './index';

let wrapped;
const props = {
  className: 'TitleClass'
};

beforeEach(() => {
  wrapped = mount(<Social {...props} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('Social', () => {
  it('renders the Social Component', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('renders the p tag that holds the social media icons', () => {
    expect(wrapped.find('p').hasClass(props.className)).toBe(true);
  });

  it('renders 3 <a> tags', () => {
    expect(wrapped.find('a').length).toEqual(3);
  });

});

describe('Social with no className', () => {
  let comp;
  beforeEach(() => {
    comp = mount(<Social />);
  });
  
  afterEach(() => {
    comp.unmount();
  });

  it('renders no className when none is given', () => {
    expect(comp.find('p').hasClass(props.className)).toBe(false);
  });

});