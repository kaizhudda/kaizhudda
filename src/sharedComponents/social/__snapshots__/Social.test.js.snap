// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social renders the Social Component 1`] = `
<Social
  className="TitleClass"
>
  <p
    className="Social TitleClass"
  >
    <a
      href="https://linkedin.com/in/kaizhudda"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-linkedin"
      />
    </a>
    <a
      href="https://github.com/kaizhudda"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-github-square"
      />
    </a>
    <a
      href="https://twitter.com/kaizhudda"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-twitter-square"
      />
    </a>
  </p>
</Social>
`;
